Submitted By:            Bruce Dubbs <bdubbs at linuxfromscratch dot org>
Updated By:              Douglas R. Reno <renodr at linuxfromscratch dot org>
Date:                    2024-11-18
Updated:                 2024-12-03
Initial Package Version: 24.8.3.2
Origin:                  Upstream
Upstream Status:         Pending
Description:             Fixes building Libreoffice with poppler-24.11.0.
Update Description:      Fixes building Libreoffice with poppler-24.12.0 by
                         adjusting a call to getImageParams.

diff -up libreoffice-24.8.2.1/sdext/source/pdfimport/xpdfwrapper/pdfioutdev_gpl.cxx.omv~ libreoffice-24.8.2.1/sdext/source/pdfimport/xpdfwrapper/pdfioutdev_gpl.cxx
--- libreoffice-24.8.2.1/sdext/source/pdfimport/xpdfwrapper/pdfioutdev_gpl.cxx.omv~	2024-10-20 19:08:49.935352230 +0200
+++ libreoffice-24.8.2.1/sdext/source/pdfimport/xpdfwrapper/pdfioutdev_gpl.cxx	2024-10-20 20:01:30.448965822 +0200
@@ -1090,7 +1090,8 @@
 
     int bitsPerComponent = 1;
     StreamColorSpaceMode csMode = streamCSNone;
-    str->getImageParams( &bitsPerComponent, &csMode );
+    bool hasAlpha = false;
+    str->getImageParams( &bitsPerComponent, &csMode, &hasAlpha );
     if( bitsPerComponent == 1 && (csMode == streamCSNone || csMode == streamCSDeviceGray) )
     {
         GfxRGB oneColor = { dblToCol( 1.0 ), dblToCol( 1.0 ), dblToCol( 1.0 ) };
@@ -1306,9 +1306,9 @@ poppler_bool PDFOutDev::tilingPatternFil
         0, nBitmapWidth * nBitmapHeight, Object(objNull)));
     auto aDecode = Object(objNull);
     std::unique_ptr<GfxImageColorMap> pRgbIdentityColorMap(new GfxImageColorMap(8, &aDecode,
-        new GfxDeviceRGBColorSpace()));
+        std::make_unique<GfxDeviceRGBColorSpace>()));
     std::unique_ptr<GfxImageColorMap> pGrayIdentityColorMap(new GfxImageColorMap(8, &aDecode,
-        new GfxDeviceGrayColorSpace()));
+        std::make_unique<GfxDeviceGrayColorSpace>()));
 
     OutputBuffer aBuf; initBuf(aBuf);
     writePng_(aBuf, pRgbStr.get(), nBitmapWidth, nBitmapHeight, pRgbIdentityColorMap.get(),

